tokio/src/task/budget.rs

```diff
@@ -39,3 +39,21 @@ pub async fn consume_budget() {
     })
     .await
 }
+
+/// Polls to see if any budget is available or not.
+///
+/// See also the usage example in the [task module](index.html#poll_budget_available).
+///
+/// This method returns:
+/// - `Poll::Pending` if the budget is depleted
+/// - `Poll::Ready(())` if there is still budget left
+#[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+pub fn poll_budget_available(cx: &mut Context<'_>) -> Poll<()> {
+    ready!(crate::trace::trace_leaf(cx));
+    if crate::runtime::coop::has_budget_remaining() {
+        Poll::Ready(())
+    } else {
+        cx.waker().wake_by_ref();
+        Poll::Pending
+    }
+}
```

Comment:
I think that the `poll_proceed` / `made_progress` API we use internally is good. If we're going to expose more of coop, then I think we should expose that API instead of inventing a new one.

But I wouldn't do both "expose coop API" and "make select coop aware" in one PR. I'd like two PRs for the changelog.

Reply:
Makes sense. I'll focus on exposing our coop API in this PR and will make select budget-aware afterwards in a follow-up PR.